docs/client_middleware_cookbook.rst

```diff
@@ -5,331 +5,95 @@
-                        self.max_retries + 1
-                    )
-
-                # Execute the request
-                response = await handler(request)
-                last_response = response
-
-                # Check if we should retry
-                if response.status not in self.retry_statuses:
-                    return response
-
-                # Don't retry if we've exhausted attempts
-                if attempt >= self.max_retries:
-                    _LOGGER.warning(
-                        "Max retries (%s) exceeded for %s",
-                        self.max_retries,
-                        request.url
-                    )
-                    return response
-
-                # Wait before retrying
-                _LOGGER.debug("Waiting %ss before retry...", delay)
-                await asyncio.sleep(delay)
-                delay *= self.backoff_factor
-
-            # Return the last response
-            return last_response
-
-Usage example:
-
-.. code-block:: python
-
-    import aiohttp
-    import asyncio
-    import logging
-    from http import HTTPStatus
-
-    _LOGGER = logging.getLogger(__name__)
-
-    RETRY_STATUSES = {
-        HTTPStatus.TOO_MANY_REQUESTS,
-        HTTPStatus.INTERNAL_SERVER_ERROR,
-        HTTPStatus.BAD_GATEWAY,
-        HTTPStatus.SERVICE_UNAVAILABLE,
-        HTTPStatus.GATEWAY_TIMEOUT
-    }
-
-    async def main():
-        # Create retry middleware with custom settings
-        retry_middleware = RetryMiddleware(
-            max_retries=3,
-            retry_statuses=RETRY_STATUSES,
-            initial_delay=0.5,
-            backoff_factor=2.0
-        )
-
-        async with aiohttp.ClientSession(middlewares=(retry_middleware,)) as session:
-            # This will automatically retry on server errors
-            async with session.get("https://httpbin.org/status/500") as resp:
-                _LOGGER.debug("Final status: %s", resp.status)
-
-    asyncio.run(main())
-
-.. _cookbook-combining-middleware:
-
-Combining Multiple Middleware
------------------------------
-
-You can combine multiple middleware to create powerful request pipelines:
-
-.. code-block:: python
-
-    import time
-    import logging
-    from aiohttp import ClientRequest, ClientResponse, ClientHandlerType
-
-    _LOGGER = logging.getLogger(__name__)
-
-    class LoggingMiddleware:
-        """Middleware that logs request timing and response status."""
-
-        async def __call__(
-            self,
-            request: ClientRequest,
-            handler: ClientHandlerType
-        ) -> ClientResponse:
-            start_time = time.monotonic()
-
-            # Log request
-            _LOGGER.debug("[REQUEST] %s %s", request.method, request.url)
-
-            # Execute request
-            response = await handler(request)
+    These can also be adjusted to handle proxy auth by modifying
+    :attr:`ClientRequest.proxy_headers` or :attr:`ClientSession.proxy_headers`.
```

Comment:
There's no proxy_headers on the session, which I guess means we need to use a middleware currently to do even basic auth on the proxy.
```suggestion
    :attr:`ClientRequest.proxy_headers`.
```